src/rnn/Solutions/bwd_s_stream.cpp

```diff
@@ -134,5 +134,94 @@ void RNNModularSingleStreamBWD::ComputeBWD(Handle& handle,
 #endif
 }
 
+void RNNDynamicModularSingleStreamBWD::ComputeBWD(Handle& handle,
+                                                  const runtimeArgsBwd& realRuntimeArgs) const
+{
+    auto layer_i = rnnDesc.nLayers;
+
+    auto seq_iterations      = rnnAlgoModules.getTimeSeqSize();
+    auto real_seq_iterations = rnnAlgoModules.getRealTimeSeqSize();
+
+    if(layer_i == 0 || seq_iterations == 0 || real_seq_iterations == 0)
+        return;
+
+    auto sequence_directions =
+        rnnDesc.dirMode == miopenRNNDirectionMode_t::miopenRNNbidirection ? 2 : 1;
+
+    const auto runtimeArgsExt = rnnAlgoModules.createRuntimeArgsExt(realRuntimeArgs);
+    const auto [real_dy,
+                temp_dy,
+                dhy,
+                dhx,
+                cx,
+                dcy,
+                dcx,
+                real_dx,
+                temp_dx,
+                w,
+                workSpace,
+                reserveSpace] = runtimeArgsExt;
```

Comment:
```suggestion
    const auto runtimeArgsExt = rnnAlgoModules.createRuntimeArgsExt(realRuntimeArgs);
    const auto& [real_dy,
                temp_dy,
                dhy,
                dhx,
                cx,
                dcy,
                dcx,
                real_dx,
                temp_dx,
                w,
                workSpace,
                reserveSpace] = runtimeArgsExt;
```
I'm not sure that the full copy of everything is needed. Reference seems reasonable in that case.
https://godbolt.org/z/afG9rsTP9

Reply:
Good point

Reply:
But this is not only a copy, the main compiler optimization to save data in register (and read only once from stack) if we have multiple usage in this function, in the moment it cost's us extra push and pop (from stack) operation per used register. But in reference case this is read from stack each data for each function.

Reply:
It depends, it does not require to be on stack - it's not volatile and it's up to compiler to decide. Reference can be a pointer, but actually it's just an alias to some entity and it can perfectly live on registers, if compiler decides that it worth living there.
If you take a look into an example - the copy is a real copy and it resides on stack.

1) `rnnAlgoModules.createRuntimeArgsExt` has always returned on stack.
2) `RuntimeArgsExt` is quite huge - 12 poninter
3) You need a readonly access
4) x86-64 has 16 GPRs, but actually ~14 or less
5) following code is quite complex, has some function calls (more GPRs wasted on class), lots of read-write local variables, etc.

I would not expect even the half of those variables can be places on registers. If it would be a tight loop and a small struct, then may be.
Btw, you can disassemble and compare 2 cases, it's rather simple.

P.S. I'm not insisting on that change, I'm just saying that the assumption may be far from reality.